dbt version updates (#9590)
fishtown-analytics/docs.getdbt.com: website/docs/docs/fusion/fusion-version-compatibility.md

---
title: "Version compatibility"
id: "fusion-version-compatibility"
description: "Understand how the dbt Fusion engine, language server, and VS Code extension versions map together, and how to verify compatible binaries."
pagination_next: null
pagination_prev: null
---

# Version compatibility <Lifecycle status="preview" />

The <Constant name="fusion_engine" />, its language server (<Term id="lsp" />), and the dbt VS Code extension work together to power local development. This page explains how their versions relate, which combinations are compatible, and how to verify you have matching binaries.

## How versioning works

> <Constant name="fusion" /> ships as a single binary that provides both the command-line interface (CLI) and the language server (LSP). Because the CLI and LSP come from the _same_ binary, they can't be version-mismatched with each other.

Reviewer: ```suggestion
<Constant name="fusion" /> ships as a single binary that provides both the command-line interface (CLI) and the language server (LSP). Because the CLI and LSP come from the _same_ binary, they always share the same version and can't be mismatched.
```

The dbt VS Code extension keeps its own release version because it follows the VS Code Marketplace release lifecycle. To make compatibility explicit, each extension release declares the range of <Constant name="fusion_engine" /> versions it supports. When the extension starts the LSP, it checks the installed <Constant name="fusion" /> version against that range and surfaces an [actionable error](#troubleshooting-version-issues) if they don't match.

| Component | Example versioning |
| --------- | ---------- |
| <Constant name="fusion_engine" /> binary (CLI + LSP) | `2.0.0-preview.N` |

> | dbt VS Code extension | THe extension version (for example, `0.36.0`), with compatible <Constant name="fusion" /> version range |

Reviewer: ```suggestion
| dbt VS Code extension | The extension version (for example, `0.36.0`), with compatible <Constant name="fusion" /> version range |
```

## Compatibility matrix

The following table maps each dbt VS Code extension version to the <Constant name="fusion_engine" /> versions it supports. If you let the extension download and manage <Constant name="fusion" /> for you, it always installs a compatible version automatically &mdash; you only need this table if you install or pin binaries manually.

| dbt VS Code extension | Min <Constant name="fusion" /> version | Max <Constant name="fusion" /> version | Checksums | Notes |
| --- | --- | --- | --- | --- |
| 0.36.x | 2.0.0-preview.90 | &mdash; | `checksums-0.36.txt` | Current |
| 0.35.x | 2.0.0-preview.80 | 2.0.0-preview.89 | `checksums-0.35.txt` | - |

A dash (-) in the **Max** column means there is no upper bound yet. Any <Constant name="fusion" /> release at or above the minimum is supported.

## Check your installed versions

> Run `dbt --version` to see the version of the <Constant name="fusion_engine" /> you have installed:

Reviewer: ```suggestion
To confirm your setup is within the supported range, check the version of the <Constant name="fusion_engine" /> you have installed:
```
